modular_skyrat/modules/moretraitoritems/code/syndiemirror.dm

```diff
@@ -0,0 +1,59 @@
+/obj/item/hhmirror/syndie
+	name = "Handheld Mirror"
+	desc = "A handheld mirror that allows you to change your looks. Reminds you of old times for some reason..."
+	icon = 'modular_skyrat/master_files/icons/obj/hhmirror.dmi'
+	icon_state = "hhmirror"
+	special_desc_requirement = EXAMINE_CHECK_SYNDICATE
+	special_desc = "A mirror manufactured by the Syndicate containing barber-nanites that can alter your hair on the spot."
+	w_class = WEIGHT_CLASS_SMALL
+
+/obj/item/storage/box/syndie_kit/chameleon/PopulateContents()
+	new /obj/item/clothing/under/chameleon(src)
+	new /obj/item/clothing/suit/chameleon(src)
+	new /obj/item/clothing/gloves/chameleon(src)
+	new /obj/item/clothing/shoes/chameleon(src)
+	new /obj/item/clothing/glasses/chameleon(src)
+	new /obj/item/clothing/head/chameleon(src)
+	new /obj/item/clothing/mask/chameleon(src)
+	new /obj/item/clothing/neck/chameleon(src)
+	new /obj/item/storage/backpack/chameleon(src)
+	new /obj/item/storage/belt/chameleon(src)
+	new /obj/item/radio/headset/chameleon(src)
+	new /obj/item/stamp/chameleon(src)
+	new /obj/item/pda/chameleon(src)
+	new /obj/item/hhmirror/syndie(src)
+	new /obj/item/dyespray(src)
+
+
+/obj/item/hhmirror/attack_self(mob/user)
+	. = ..()
+	if(.)
+		return
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+			//see code/modules/mob/dead/new_player/preferences.dm at approx line 545 for comments!
+			//this is largely copypasted from there.
+
+			//handle facial hair (if necessary)
+		if(H.gender != FEMALE)
+			var/new_style = input(user, "Select a facial hair style", "Grooming")  as null|anything in GLOB.facial_hairstyles_list
+			if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+				return	//no tele-grooming
+			if(new_style)
+				H.facial_hairstyle = new_style
+		else //look who cant remove if else's
+			//H.facial_hairstyle = "Shaved"
+			var/new_style = input(user, "Select a facial hair style", "Grooming")  as null|anything in GLOB.facial_hairstyles_list
+			if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+				return	//no tele-grooming
+			if(new_style)
+				H.facial_hairstyle = new_style
+
+			//handle normal hair
+		var/new_style = input(user, "Select a hair style", "Grooming")  as null|anything in GLOB.hairstyles_list
+		if(!user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+			return	//no tele-grooming
+		if(new_style)
+			H.hairstyle = new_style
+
+		H.update_hair()
```

Comment:
In general for this proc, just take a look at what I did for the barber's scissors and inspire yourself of that instead of this.